include/tunnler/room_member.h

```diff
@@ -149,6 +149,12 @@ class RoomMember final {
      */
     void HandleJoinPacket(const RakNet::Packet* packet);
 
+    /**
+     * Closes the connection and sets the variables to a clean state for a new connection.
+     * Must be called, when connection to server is closed.
+     */
+    void CloseConnection();
```

Comment:
I thought long about that, and I still don't like both solutions. I don't like to be shutdown in CloseConnection but everytime shutdown is called CloseConnection should be called...